src/arc_utilities/zlib_helpers.cpp

```diff
@@ -130,6 +132,30 @@ namespace ZlibHelpers
 
     std::vector<uint8_t> LoadFromFileAndDecompress(const std::string& path)
     {
+        if (!boost::filesystem::is_regular_file(path))
+        {
+            if (boost::filesystem::is_regular_file(path + "." + std::to_string(0)))
+            {
+                std::cerr << "ZLibHelpers::LoadFromFileAndDecompress(): " << path << " is not a regular file. Loading from individual slices." << std::endl;
+
+                std::vector<uint8_t> decompressed;
+                int file_idx = 0;
+                std::string slice_path = path + "." + std::to_string(file_idx);
+                do
+                {
+                    const auto slice_bytes = LoadFromFileAndDecompress(slice_path);
+                    std::cerr << "Reading bytes [" << decompressed.size() << ", " << decompressed.size() + slice_bytes.size() << ") from " << slice_path << std::endl;
+                    decompressed.reserve(decompressed.size() + slice_bytes.size());
+                    decompressed.insert(decompressed.end(), slice_bytes.cbegin(), slice_bytes.cend());
+
+                    ++file_idx;
+                    slice_path = path + "." + std::to_string(file_idx);
+                }
+                while (boost::filesystem::is_regular_file(slice_path));
+                return decompressed;
+            }
+        }
```

Comment:
Unhandled option.
Presumable `if(!is_regular_file(path))` and `if(!is_regular_file(path + ".0"))` you want to abort. Right?

Reply:
Not actually unhandled, but not explicitly handled. Right now it eventually throws an exception later in the code. I'll explicitly handle it for easier code maintenance later.